File: app/javascript/channels/timer_channel.js
import consumer from "./consumer"
import ClockTicker from "./clock_ticker"

const user_id = $("meta[name=current-user]").attr("content")
const clock = new ClockTicker()

const sub = consumer.subscriptions.create(
  { channel: "TimerChannel", user_id: user_id },
  {
    connected() {
      console.log(`TimerChannel(${user_id}) connected`, { sub })
      this.perform("get_timer")
    },

    received(timer) {
      console.log(`TimerChannel(${user_id}) received`, { timer })
      clock.new_timer(timer)
    },

    disconnected(info) {
      console.log(`TimerChannel(${user_id}) disconnected`, { info })
    },
  }
)

File: app/channels/timer_channel.rb
class TimerChannel < ApplicationCable::Channel
  def subscribed
    user_id = params[:user_id].to_i
    if user_id == current_user.id
      stream_for params[:user_id]
    else
      reject
    end
  end

  def get_timer(data)
    timer = Timer.find_by(user_id: current_user.id)
    TimerChannel.broadcast_to(current_user.id, timer)
  end
end
